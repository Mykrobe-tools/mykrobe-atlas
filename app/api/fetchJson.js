/* @flow */

import fetch from 'isomorphic-fetch';
import type { JSendType } from '../types/JSendType';
import type { UserType } from '../types/UserTypes';
import * as AuthActions from '../actions/AuthActions';

import store from '../store/store';

/**
 * Request and parse a JSend resource from the API
 * Adds Authorization token if stored in cookie
 * Resolves with json object or rejects with JSend or request error
 * Uses dispatch to send action for any unauthorized request
 * @param  {dispatch} Function
 * @param  {string} url
 * @param  {any={}} options
 * @returns Promise
 */

export default (url: string, options: any = {}): Promise<any> => {
  if (!options.headers) {
    options.headers = {};
  }
  options.headers = {
    ...options.headers,
    Accept: 'application/json',
  };
  const user: UserType = store.getState().auth.user;
  if (user && user.token) {
    options.headers = {
      ...options.headers,
      Authorization: `Bearer ${user.token}`,
    };
  }
  console.log('fetch options', options);
  return fetch(url, options)
    .then(response => {
      if (response.ok) {
        return response.json().then((jsend: JSendType) => {
          const { status, message, data } = jsend;
          if (!status) {
            const json = JSON.stringify(jsend, null, 2);
            return Promise.reject({
              status: response.status,
              statusText: `Invalid JSend response ${json}`,
            });
          } else if (status === 'success') {
            return Promise.resolve(data);
          }
          return Promise.reject({
            status: response.status,
            statusText: message,
          });
        });
      } else {
        if (response.status === 401) {
          AuthActions.signOut()(store.dispatch);
        }
        return Promise.reject({
          status: response.status,
          statusText: response.statusText,
        });
      }
    })
    .catch(error => {
      // An error we generated
      if (error.statusText) {
        return Promise.reject(error);
      }
      // An error from underlying network, e.g. failed preflight check
      return Promise.reject({
        statusText: error.message,
      });
    });
};
