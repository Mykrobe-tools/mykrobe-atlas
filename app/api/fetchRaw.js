/* @flow */

import fetch from 'isomorphic-fetch';
import type { UserType } from '../types/UserTypes';

import store from '../store/store';

/**
 * Adds Authorization token
 */

export default (url: string, options: any = {}): Promise<any> => {
  if (!options.headers) {
    options.headers = {};
  }
  options.headers = {
    ...options.headers,
    'Accept': 'application/json'
  };
  const user: UserType = store.getState().auth.user;
  if (user && user.token) {
    options.headers = {
      ...options.headers,
      'Authorization': `Bearer ${user.token}`
    };
  }
  console.log('fetch options', options);
  return fetch(url, options);
};
