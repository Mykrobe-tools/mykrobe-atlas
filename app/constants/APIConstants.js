/* @flow */

export const BASE_URL: string =
  process.env.NODE_ENV === 'production'
    ? (process.env.API_URL: any)
    : 'https://api.atlas-dev.makeandship.com';

// export const BASE_URL = 'https://localhost:3001';
// export const BASE_URL = 'http://13.69.243.89:8000';

export const USER_COOKIE_NAME = 'api.atlas.makeandship.com';
