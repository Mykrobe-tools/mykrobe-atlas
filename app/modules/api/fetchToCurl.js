/* @flow */

export function fetchToCurl(url: string, options: any = {}) {
  const { method, headers, body } = options;
  const components = ['curl', `'${url}'`];
  if (method) {
    components.push('-X');
    components.push(method);
  }
  for (let key in headers) {
    const value = headers[key];
    components.push('-H');
    components.push(`'${key}: ${value}'`);
  }
  if (body) {
    components.push('--data-binary');
    components.push(`'${body}'`);
  }
  const curl = components.join(' ');
  return curl;
}
