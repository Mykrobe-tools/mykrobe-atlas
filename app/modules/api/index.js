/* @flow */

export { FETCH_JSON, fetchJsonMiddleware } from './fetchJsonMiddleware';

export { fetchJson, fetchJsonWithToken } from './fetchJson';
