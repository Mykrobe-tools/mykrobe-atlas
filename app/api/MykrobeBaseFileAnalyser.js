import { EventEmitter } from 'events';
import * as TargetConstants from 'constants/TargetConstants';
import MykrobeJsonTransformer from './MykrobeJsonTransformer';

class MykrobeBaseFileAnalyser extends EventEmitter {
  constructor(targetConfig) {
    super();
    this.targetConfig = targetConfig;
  }

  extensionForFileName(fileName) {
    const extension = fileName.substr(fileName.lastIndexOf('.'));
    return extension.toLowerCase();
  }

  analyseFile(file) {
    this.cancel();
    const extension = this.extensionForFileName(file.name);
    if ('.json' === extension) {
      return this.analyseJsonFile(file);
    }
    else if (['.bam', '.gz', '.fastq'].indexOf(extension) !== -1) {
      return this.analyseBinaryFile(file);
    }
    else {
      this.failWithError(`Can only process files with extension: .json, .bam, .gz, .fastq - not ${extension}`);
      return this;
    }
  }

  failWithError(err) {
    setTimeout(() => {
      this.emit('error', {
        description: `Processing failed with error: ${err}`
      });
    }, 0);
  }

  doneWithJsonString(jsonString) {
    const transformer = new MykrobeJsonTransformer();
    transformer.transform(jsonString).then((result) => {
      const {json, transformed} = result;
      console.log('json', json);
      console.log('transformed', transformed);
      this.emit('done', result);
    })
    .catch((err) => {
      this.failWithError(err);
    });
  }

  analyseJsonFile(file) {
    const reader = new FileReader();

    reader.onload = (e) => {
      const dataString = reader.result;
      this.doneWithJsonString(dataString);
    };

    reader.onerror = (e) => {
      this.failWithError(`FileReader failed with error code ${e.target.error.code}`);
    };

    reader.readAsText(file);
    return this;
  }

  analyseBinaryFile(file) {
    console.log('analyseBinaryFile', file);
  }

  cancel() {
  }
}

export default MykrobeBaseFileAnalyser;
