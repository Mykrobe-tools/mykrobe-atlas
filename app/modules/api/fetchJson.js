/* @flow */

import type { JSendType } from '../../types/JSendType';

import {
  getAccessToken,
  signOut,
} from 'makeandship-js-common/src/modules/auth';
import {
  showNotification,
  NotificationCategories,
} from 'makeandship-js-common/src/modules/notifications';

import { fetchToCurl } from './fetchToCurl';

/**
 * Request and parse a JSend resource from the API
 * Adds Authorization token if stored in cookie
 * Resolves with json object or rejects with JSend or request error
 * Uses dispatch to send action for any unauthorized request
 * @param  {dispatch} Function
 * @param  {string} url
 * @param  {any={}} options
 * @returns Promise
 */

export function fetchJson(url: string, options: any = {}) {
  return (dispatch: Function, getState: Function) => {
    const state = getState();
    const token = getAccessToken(state);
    return fetchJsonWithTokenAndDispatch(url, options, token, dispatch);
  };
}

// TODO: this is a workaround for services which currently do not have redux integration

export function fetchJsonWithToken(
  url: string,
  options: any = {},
  token?: string
) {
  return fetchJsonWithTokenAndDispatch(url, options, token);
}

function fetchJsonWithTokenAndDispatch(
  url: string,
  options: any = {},
  token?: string,
  dispatch?: Function
) {
  if (!options.headers) {
    options.headers = {};
  }
  options.headers = {
    ...options.headers,
    Accept: 'application/json',
  };
  if (token) {
    options.headers = {
      ...options.headers,
      Authorization: `Bearer ${token}`,
    };
  }
  console.log(`fetch ${url} options`, options);
  const curl = fetchToCurl(url, options);
  console.log(curl);
  return fetch(url, options)
    .then(response => {
      if (response.ok) {
        return response.json().then((jsend: JSendType) => {
          const { status, message, data } = jsend;
          if (!status) {
            const json = JSON.stringify(jsend, null, 2);
            return Promise.reject({
              status: response.status,
              statusText: `Invalid JSend response ${json}`,
            });
          } else if (status === 'success') {
            console.log('data', JSON.stringify(data, null, 2));
            return Promise.resolve(data);
          }
          return Promise.reject({
            status: response.status,
            statusText: message || data,
          });
        });
      } else {
        if (token && response.status === 401) {
          dispatch && dispatch(signOut());
        }
        return Promise.reject({
          status: response.status,
          statusText: response.statusText,
        });
      }
    })
    .catch(error => {
      dispatch &&
        dispatch(
          showNotification({
            category: NotificationCategories.ERROR,
            content: error.statusText || error.message,
          })
        );
      // An error we generated
      if (error.statusText) {
        return Promise.reject(error);
      }
      // An error from underlying network, e.g. failed preflight check
      return Promise.reject({
        statusText: error.message,
      });
    });
}
