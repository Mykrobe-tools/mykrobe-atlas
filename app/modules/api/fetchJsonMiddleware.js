/* @flow */

import 'isomorphic-fetch';

import type { JSendType } from '../../types/JSendType';

// TODO: make token access more generic
import {
  getAuthToken,
  signOut,
  AUTH_SIGNOUT,
  AUTH_SIGNOUT_SUCCESS,
} from '../auth/auth';

import {
  showNotification,
  NotificationCategories,
} from '../notifications/notifications';

import { fetchToCurl } from './fetchToCurl';

export const FETCH_JSON = '@@fetch-json-middleware/FETCH_JSON';

export class FetchJsonError extends Error {
  constructor(status, statusText, response) {
    super();
    this.name = 'FetchJsonError';
    this.status = status;
    this.statusText = statusText;
    this.response = response;
    this.message = `${status} - ${statusText}`;
  }
}

const isFetchJson = action => {
  return action.hasOwnProperty && action.hasOwnProperty(FETCH_JSON);
};

const normalizeTypes = types =>
  types.map(type => {
    if (typeof type === 'string') {
      return {
        type,
      };
    }
    return type;
  });

export const fetchJsonMiddleware = store => next => action => {
  // Do not process actions without [FETCH_JSON] property
  if (!isFetchJson(action)) {
    return next(action);
  }

  const fetchJsonParameters = action[FETCH_JSON];

  const { url, options = {}, types, debug = false } = fetchJsonParameters;

  const [REQUEST, SUCCESS, FAILURE] = normalizeTypes(types);

  next(REQUEST);

  const state = store.getState();
  const token = getAuthToken(state);

  if (!options.headers) {
    options.headers = {};
  }
  options.headers = {
    ...options.headers,
    Accept: 'application/json',
  };
  if (token) {
    options.headers = {
      ...options.headers,
      Authorization: `Bearer ${token}`,
    };
  }

  if (debug) {
    console.log(`fetch ${url} options`, options);
    const curl = fetchToCurl(url, options);
    console.log(curl);
  }

  return (async () => {
    try {
      const response = await fetch(url, options);

      if (!response.ok) {
        if (token && response.status === 401) {
          await store.dispatch(signOut());
        }
        throw new FetchJsonError(
          response.status,
          response.statusText,
          response
        );
      }

      let jsend: JSendType;

      try {
        jsend = await response.json();
      } catch (error) {
        throw new FetchJsonError(response.status, error.message, response);
      }

      const { status, message, data } = jsend;

      if (!status) {
        const json = JSON.stringify(jsend, null, 2);
        throw new FetchJsonError(
          response.status,
          `Invalid JSend response - ${json}`,
          response
        );
      }

      if (status !== 'success') {
        throw new FetchJsonError(response.status, message || data, response);
      }

      if (debug) {
        console.log('data', JSON.stringify(data, null, 2));
      }

      next({ ...SUCCESS, payload: data });
      return data;
    } catch (error) {
      next(FAILURE);
      const content = error.message || error.statusText;
      await store.dispatch(
        showNotification({
          category: NotificationCategories.ERROR,
          content,
        })
      );
      throw error;
    }
  })();
};
