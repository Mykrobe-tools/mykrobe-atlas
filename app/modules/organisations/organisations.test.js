/* @flow */

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';

import { fetchJsonMiddleware } from '../api';
import { API_URL } from '../../constants/APIConstants.js';

import reducer, { initialState, requestOrganisations } from './organisations';

const createMockStore = configureMockStore([thunk, fetchJsonMiddleware]);
const data = require('../../../test/__fixtures__/api/organisations.json');

describe('organisations module', () => {
  const store = createMockStore(initialState);
  let mockState;

  beforeEach(() => {
    store.clearActions();
    nock.cleanAll();
  });

  it('should handle "requestOrganisations" action', async () => {
    nock(API_URL)
      .get('/organisations')
      .reply(200, data);
    const payload = await store.dispatch(requestOrganisations());
    const dispatchedActions = store.getActions();
    dispatchedActions.forEach(dispatchedAction => {
      mockState = reducer(mockState, dispatchedAction);
    });
    expect(dispatchedActions).toMatchSnapshot();
    expect(mockState).toMatchSnapshot();
    expect(payload).toEqual(data.data);
  });
});
