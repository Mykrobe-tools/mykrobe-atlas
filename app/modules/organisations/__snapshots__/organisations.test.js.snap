// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisations module should handle "requestOrganisations" action 1`] = `
Array [
  Object {
    "type": "organisations/organisations/REQUEST_ORGANISATIONS",
  },
  Object {
    "payload": Array [
      Object {
        "id": "591ab833d1f7fd000e5083a3",
        "name": "Pacocha, Harris and Metz",
        "template": "",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a4",
        "name": "Daugherty and Sons",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a5",
        "name": "Kreiger - Kovacek",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a6",
        "name": "Fay - Walsh!",
        "template": "MGIT",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a7",
        "name": "Littel Inc",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a8",
        "name": "Morissette, Abshire and Pfeffer",
      },
      Object {
        "id": "591ab833d1f7fd000e5083a9",
        "name": "Weber, Stanton and Hagenes",
      },
      Object {
        "id": "591ab833d1f7fd000e5083aa",
        "name": "Lang and Sons",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ab",
        "name": "Fritsch, Hyatt and Muller",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ac",
        "name": "Wunsch LLC",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ad",
        "name": "Weissnat - Parisian",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ae",
        "name": "Hartmann - Kihn",
      },
      Object {
        "id": "591ab833d1f7fd000e5083af",
        "name": "Goodwin Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b0",
        "name": "Kutch Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b1",
        "name": "Kiehn LLC",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b2",
        "name": "Leannon - Mertz",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b3",
        "name": "Ebert Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b4",
        "name": "Metz and Sons",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b5",
        "name": "Blanda Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b6",
        "name": "Hilpert Inc",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b7",
        "name": "Bode, Pfeffer and Predovic",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b8",
        "name": "Renner Inc",
      },
      Object {
        "id": "591ab833d1f7fd000e5083b9",
        "name": "Effertz - Glover",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ba",
        "name": "Farrell, Christiansen and Marvin",
      },
      Object {
        "id": "591ab833d1f7fd000e5083bb",
        "name": "Hahn Inc",
      },
      Object {
        "id": "591ab833d1f7fd000e5083bc",
        "name": "Olson, Blick and Feeney",
      },
      Object {
        "id": "591ab833d1f7fd000e5083bd",
        "name": "Von, Stokes and Strosin",
      },
      Object {
        "id": "591ab833d1f7fd000e5083be",
        "name": "Johnson, Murazik and Gusikowski",
      },
      Object {
        "id": "591ab833d1f7fd000e5083bf",
        "name": "Nolan, Kuhn and Maggio",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c0",
        "name": "Will - Stark",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c1",
        "name": "Parker - Armstrong",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c2",
        "name": "Baumbach, Wolff and Leuschke",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c3",
        "name": "Ward - Auer",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c4",
        "name": "Smitham LLC",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c5",
        "name": "Effertz Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c6",
        "name": "Anderson - Bechtelar",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c7",
        "name": "Berge Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c8",
        "name": "Rowe Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083c9",
        "name": "Adams Inc",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ca",
        "name": "Adams, Shields and Bernier",
      },
      Object {
        "id": "591ab833d1f7fd000e5083cb",
        "name": "Steuber, Huels and Altenwerth",
      },
      Object {
        "id": "591ab833d1f7fd000e5083cc",
        "name": "Altenwerth, Cronin and Quigley",
      },
      Object {
        "id": "591ab833d1f7fd000e5083cd",
        "name": "Zemlak - Strosin",
      },
      Object {
        "id": "591ab833d1f7fd000e5083ce",
        "name": "Halvorson - Ernser",
      },
      Object {
        "id": "591ab833d1f7fd000e5083cf",
        "name": "Dooley, Nolan and Kulas",
      },
      Object {
        "id": "591ab833d1f7fd000e5083d0",
        "name": "Heller and Sons",
      },
      Object {
        "id": "591ab833d1f7fd000e5083d1",
        "name": "Green and Sons",
      },
      Object {
        "id": "591ab833d1f7fd000e5083d2",
        "name": "Dietrich, Halvorson and Witting",
      },
      Object {
        "id": "591ab833d1f7fd000e5083d3",
        "name": "Casper Group",
      },
      Object {
        "id": "591ab833d1f7fd000e5083d4",
        "name": "Robel - Schowalter",
      },
    ],
    "type": "organisations/organisations/REQUEST_ORGANISATIONS_SUCCESS",
  },
]
`;

exports[`organisations module should handle "requestOrganisations" action 2`] = `
Object {
  "error": undefined,
  "isFetching": false,
  "organisations": Array [
    Object {
      "id": "591ab833d1f7fd000e5083a3",
      "name": "Pacocha, Harris and Metz",
      "template": "",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a4",
      "name": "Daugherty and Sons",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a5",
      "name": "Kreiger - Kovacek",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a6",
      "name": "Fay - Walsh!",
      "template": "MGIT",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a7",
      "name": "Littel Inc",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a8",
      "name": "Morissette, Abshire and Pfeffer",
    },
    Object {
      "id": "591ab833d1f7fd000e5083a9",
      "name": "Weber, Stanton and Hagenes",
    },
    Object {
      "id": "591ab833d1f7fd000e5083aa",
      "name": "Lang and Sons",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ab",
      "name": "Fritsch, Hyatt and Muller",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ac",
      "name": "Wunsch LLC",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ad",
      "name": "Weissnat - Parisian",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ae",
      "name": "Hartmann - Kihn",
    },
    Object {
      "id": "591ab833d1f7fd000e5083af",
      "name": "Goodwin Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b0",
      "name": "Kutch Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b1",
      "name": "Kiehn LLC",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b2",
      "name": "Leannon - Mertz",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b3",
      "name": "Ebert Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b4",
      "name": "Metz and Sons",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b5",
      "name": "Blanda Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b6",
      "name": "Hilpert Inc",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b7",
      "name": "Bode, Pfeffer and Predovic",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b8",
      "name": "Renner Inc",
    },
    Object {
      "id": "591ab833d1f7fd000e5083b9",
      "name": "Effertz - Glover",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ba",
      "name": "Farrell, Christiansen and Marvin",
    },
    Object {
      "id": "591ab833d1f7fd000e5083bb",
      "name": "Hahn Inc",
    },
    Object {
      "id": "591ab833d1f7fd000e5083bc",
      "name": "Olson, Blick and Feeney",
    },
    Object {
      "id": "591ab833d1f7fd000e5083bd",
      "name": "Von, Stokes and Strosin",
    },
    Object {
      "id": "591ab833d1f7fd000e5083be",
      "name": "Johnson, Murazik and Gusikowski",
    },
    Object {
      "id": "591ab833d1f7fd000e5083bf",
      "name": "Nolan, Kuhn and Maggio",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c0",
      "name": "Will - Stark",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c1",
      "name": "Parker - Armstrong",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c2",
      "name": "Baumbach, Wolff and Leuschke",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c3",
      "name": "Ward - Auer",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c4",
      "name": "Smitham LLC",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c5",
      "name": "Effertz Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c6",
      "name": "Anderson - Bechtelar",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c7",
      "name": "Berge Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c8",
      "name": "Rowe Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083c9",
      "name": "Adams Inc",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ca",
      "name": "Adams, Shields and Bernier",
    },
    Object {
      "id": "591ab833d1f7fd000e5083cb",
      "name": "Steuber, Huels and Altenwerth",
    },
    Object {
      "id": "591ab833d1f7fd000e5083cc",
      "name": "Altenwerth, Cronin and Quigley",
    },
    Object {
      "id": "591ab833d1f7fd000e5083cd",
      "name": "Zemlak - Strosin",
    },
    Object {
      "id": "591ab833d1f7fd000e5083ce",
      "name": "Halvorson - Ernser",
    },
    Object {
      "id": "591ab833d1f7fd000e5083cf",
      "name": "Dooley, Nolan and Kulas",
    },
    Object {
      "id": "591ab833d1f7fd000e5083d0",
      "name": "Heller and Sons",
    },
    Object {
      "id": "591ab833d1f7fd000e5083d1",
      "name": "Green and Sons",
    },
    Object {
      "id": "591ab833d1f7fd000e5083d2",
      "name": "Dietrich, Halvorson and Witting",
    },
    Object {
      "id": "591ab833d1f7fd000e5083d3",
      "name": "Casper Group",
    },
    Object {
      "id": "591ab833d1f7fd000e5083d4",
      "name": "Robel - Schowalter",
    },
  ],
}
`;
